### ISSUE TITLE
[Checklist] Design checklist descriptions need to be updated

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#4139]
This arose from the entropy_src D1/V1 combined review meeting. The description for these items are really terse to the point that its is not clear what exactly needs to be done to achieve them. Filing an issue to track the required changes. 

D1: 

- [ ] CLKRST_CONNECTED
This really pertains to blocks with multiple clocks and resets. 

- [ ] IP_TOP
Explain what needs to be done to "meet comportability requirements". Perhaps change title to `IS_COMPORTABLE`, since 'top level IP exists` is really already covered by `FUNC_IMPLEMENTED`. 

- [ ] IP_INSTAITIABLE
Add link to a user guide (or expand inline) that explains how to do this (temporarily instantiating the block in the top to see topgen / intg testing is not bombing). 

@martin-lueker @mwbranstad @senelson7 @tunghoang290780 
It will be great if you all could go through the rest of the checklists (D2,3, V2,3) and document what else needs more clarifications in this issue tracker, so that we can address them all in one go.

**Comment by sriyerg** [comment]
Here's the path to the checklists for convenience:
https://docs.opentitan.org/doc/project/checklist/

**Comment by tjaychen** [comment]
i think @sriyerg and @msfschaffner had a discussion especially on the instantiable box. I can't remember the complete details now, but I believe our checklist has already been updated to reflect that.

**Comment by msfschaffner** [comment]
Yeah for the `IP_INSTANTIABLE` one we've already made an update, so I think that one we can check off.
For the other two I can add a little bit more decription.

**Comment by vogelpi** [comment]
I am closing this issue as it seems no longer relevant. Some of the checklist items have been explained in more details, and for others we have now linked additional documentation.
